qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -119,4 +125,25 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
+        y = self._convert_categorical(y)
         return ClassifierMixin.score(self, X, y, sample_weight)
+
+    def _convert_categorical(self, y: np.ndarray) -> np.ndarray:
+        """Converts categorical target data using label or one-hot encoding."""
+        y = np.array(y)
+
+        # Test if data is numeric
+        try:
+            y.astype(float)
```

Comment:
Good point, the reason why I did it this way was in case of any edge cases where, for example,  the user may have read data from a file and it stayed in string format. I really just wasn't sure what the best way was to handle all the different numpy data types so I figured seeing if it could be converted to a number was a catch-all way to check for categorical data but I am happy going your way, it is definitely more clear.